packages/component/src/SendBox/ConnectivityStatus.js

```diff
@@ -18,19 +18,21 @@ class connectConnectivityAlert extends React.Component {
     }
 
     if ( status === 'uninitialized' || ( status === 'connected' && timeoutCompleted === false && connectingSlow === false ) ) {
-      this.timerId = setTimeout( () => {
-        if ( status === 'connected' ) {
-          this.props.dispatch( { type: 'DIRECT_LINE/CONNECT_TIMEOUT_COMPLETE' } );
-        }
+      if ( status === 'connected' ) {
+        this.timerId = this.dispatchConnectivityTimeout();
         this.timerId = null;
```

Comment:
setTimeout is async so this line will immediately be executed and overwrite the timer id you just stored in `this.timerId`. Consider setting `this.timerId` to null inside of the `setTimeout()` callback